src/bot/mtproto-api/bot/scripts/sms-auth/auth-handler.ts

```diff
@@ -1,23 +1,22 @@
 // @ts-expect-error no types for this package
 import input from "input";
 import dotenv from "dotenv";
+import { createClient, SupabaseClient } from "@supabase/supabase-js";
 import { BaseMtProto } from "./base-mtproto";
 import { Context } from "../../../../../types";
 import { logger } from "../../../../../utils/logger";
-import { GithubStorage } from "../../../../../adapters/github/storage-layer";
-import { Octokit } from "@octokit/rest";
 dotenv.config();
 
 /**
  * The account holder must run this script (to my knowledge only once) to login,
  * this will give us the necessary session information to login in the future.
  */
 export class AuthHandler {
-  private _github: GithubStorage | undefined;
+  private _supabase: SupabaseClient | null = null;
   private _env = {
-    TELEGRAM_API_HASH: "",
+    TELEGRAM_API_HASH: null as string | null,
     TELEGRAM_APP_ID: 0,
-    TELEGRAM_BOT_TOKEN: "",
+    TELEGRAM_BOT_TOKEN: null as string | null,
   };
```

Comment:
It may be more concise to do this:

```
private _supabase: SupabaseClient;
```

etc

Reply:
Because you dislike empty strings I need to type it as `null` or `undefined` otherwise lots of errors happen and the reason why is the TypeBox types for these in `env` are never `null` or `undefined` so I need to cast it messily.

Empty strings here are safe because I perform checks on the variables.

Reply:
Im pretty sure my suggestion is valid to init it as undefined but its still type safe

Reply:
Yeah but we'd need to also type it as `undefined`, it's not `null` that's the problem here it's TypeBox static decodes the env as `string` so if we make it anything else we need to A) change the typebox type and subsequently use `if(!str){}` all through the codebase and retype fns for `undefined|null` across the board. if it was `null` then TypeBox would allow it when we use `Type.Decode()` which it should never do as it should only ever be `string`.

Hope this makes sense.

Reply:
> may be more concise to do this:

I can do it with the supabase prop no problem, sorry if not clear. `this._env` not so simple.

Reply:
You dont need to explicitly type uninitialized variables as undefined.

Reply:
That's true but TypeBox is the controller of the type for context.env and we don't want TypeBox to allow null or undefined ever so it's not included in the schema, in this way it enforces it exists through schema validation at runtime and uses StaticDecode to produce the intended types statically. 

We control the type for the supabase prop so I'll make the change to that. If it's still unclear I'll produce QA screenshots tomorrow and show you what I mean. 

Perhaps whilefoo or mentlegen can explain it better than I

Reply:
62ac576 - resolved

Implicit `undefined` and no explicit casting so that when used TS infers the types to be non null through the sanity checks